cpp/src/arrow/dataset/file_base.h

```diff
@@ -196,6 +200,12 @@ class ARROW_DS_EXPORT FileFragment : public Fragment,
 
   const FileSource& source() const { return source_; }
   const std::shared_ptr<FileFormat>& format() const { return format_; }
+  const int64_t start_byte() const { return start_byte_; }
+  const int64_t end_byte() const { return end_byte_; }
+  void set_bounds(int64_t start, int64_t end) {
```

Comment:
Why not let this configured as part of the `CsvFragmentScanOptions` or something?

Reply:
CsvFragmentScanOptions is global to the entire scan, so I don't think it makes sense to store fragment-specific things there.

The ability to slice a fragment is more or less local to this fragment. I.e. you could have multiple fragments in the same dataset sliced in different ways.

Reply:
I don't think this is specific to CSV either.  ARROW-17159 at least seems to suggest that it is desired for Parquet but @zhztheplayer might be in a better place to justify why that and not row group indices.  I think the goal though is to have a format-agnostic ability to slice fragments.